corehq/apps/domain/views/settings.py

```diff
@@ -582,15 +584,28 @@ def post(self, request, *args, **kwargs):
                 messages.error(request, _(e.message))
                 return HttpResponseRedirect(request.path_info)
         response = super().post(request, *args, **kwargs)
-        uidb64 = kwargs.get('uidb64')
-        uid = urlsafe_base64_decode(uidb64)
-        user = User.objects.get(pk=uid)
-        couch_user = CouchUser.from_django_user(user)
-        clear_login_attempts(couch_user)
-        domain = couch_user.domain if couch_user.is_commcare_user() else None
-        log_user_change(by_domain=None, for_domain=domain, couch_user=couch_user, changed_by_user=couch_user,
-                        changed_via=USER_CHANGE_VIA_WEB, change_messages=UserChangeMessage.password_reset(),
-                        by_domain_required_for_log=False, for_domain_required_for_log=False)
+
+        if self.get_context_data().get('form').is_valid():
```

Comment:
I think you can also move this block inside `form_valid`, which is what's called by `post` to perform the requisite actions